### PR TITLE
add login functionality for g2a

### PR DESCRIPTION
**Pull request by emraan004**

### CODE REVIEW

File: openedx/features/partners/views.py [hunk 1]
```diff
@@ -38,3 +38,22 @@ def register_user(request, slug):
...
+    """
+    This is general login view, for users of all partners
+    :param request.
+    :param slug: partner slug
+    :return: JsonResponse object with success/error message
+    """
+    partner = get_object_or_404(Partner, slug=slug)
+
+    try:
+        views = import_module(PARTNERS_VIEW_FRMT.format(slug=partner.slug))
```
Comment (zeshanarifios) [inline openedx/features/partners/views.py]: How about writing a single method to import module dynamically and raise `raise Http404('Your partner is not properly registered')` if got ImportError. and reuse it in all the views..

File: openedx/features/partners/give2asia/views.py [hunk 2]
```diff
@@ -250,3 +253,28 @@ def create_account_with_params_custom(request, params, partner):
         AUDIT_LOG.info(u"Login success on new account creation - {0}".format(new_user.username))
 
     return new_user
+
+
+class LoginSessionViewG2A(LoginSessionViewCustom):
+
```
Comment (zeshanarifios) [inline openedx/features/partners/give2asia/views.py]: please remove this line

File: openedx/features/partners/views.py [hunk 3]
```diff
@@ -38,3 +38,22 @@ def register_user(request, slug):
...
     except ImportError:
         raise Http404('Your partner is not properly registered')
+
+
+@require_http_methods(["POST"])
+@sensitive_post_parameters('password')
+def login_user(request, slug):
+    """
+    This is general login view, for users of all partners
+    :param request.
```
Comment (zahid-arbisoft) [inline openedx/features/partners/views.py]: Lets make param doc consistent, we can update it to `:param request: The Django request.`

Comment (emraan004) [reply openedx/features/partners/views.py]: Sure

**Pull request merged**